sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -495,20 +497,37 @@ export class AssessmentController {
     programYearTotal: ProgramYearTotal,
     output: VerifyAssessmentCalculationOrderJobOutDTO,
   ): void {
-    const PROGRAM_YEAR_TOTAL = "programYearTotal";
     // Create the dynamic variables to be outputted.
-    programYearTotal.awardTotals.forEach((award) => {
+    this.formulateDynamicVariables(programYearTotal.awardTotals, output);
+    this.formulateDynamicVariables(
+      programYearTotal.workflowOutputTotals,
+      output,
+    );
+  }
+
+  /**
+   * Create a new dynamic output variable for each award and offering intensity for both part-time and full-time.
+   * Create a new dynamic output variable for each workflow output.
+   * Each variable is prefixed with 'programYearTotal' and then concatenated
+   * with the offering intensity as 'FullTime'/'PartTime' and the award code.
+   * @example
+   * programYearTotalFullTimeBooksAndSuppliesCost: 150
+   * @param total awards/contributions to be added to the output.
+   * @param output output to receive the dynamic property.
+   */
+  private formulateDynamicVariables(
+    total: AwardTotal[] | ProgramYearWorkflowOutputTotal[],
```

Comment:
Instead of juggling with two types, we can create a generic type as below.
```ts
export interface ProgramYearOutputTotal<T> {
  offeringIntensity: OfferingIntensity;
  output: T;
  total: number;
}
```